cmd/bootnode/bootnode.go

```diff
@@ -86,113 +84,80 @@ func Run(ctx context.Context, config Config) error {
+			_, _ = w.Write([]byte(localEnode.Node().String()))
 		})
+		server := http.Server{Addr: config.HTTPAddr, Handler: mux, ReadHeaderTimeout: time.Second}
+		serverErr <- server.ListenAndServe()
+	}()
 
-		p2p.RegisterConnectionLogger(tcpNode, nil)
-
-		// Reservations are valid for 30min (github.com/libp2p/go-libp2p/p2p/protocol/circuitv2/relay/constraints.go:14)
-		relayResources := relay.DefaultResources()
-		relayResources.Limit.Data = 32 * (1 << 20) // 32MB
-		relayResources.MaxReservationsPerPeer = config.MaxResPerPeer
-		relayResources.MaxReservationsPerIP = config.MaxResPerPeer
-		relayResources.MaxReservations = config.MaxConns
-		relayResources.MaxCircuits = config.MaxResPerPeer
-
-		relayService, err := relay.New(tcpNode, relay.WithResources(relayResources))
-		if err != nil {
-			p2pErr <- err
+	go func() {
+		if config.MonitoringAddr == "" {
 			return
 		}
 
-		logP2P = func() {
-			peers := make(map[peer.ID]bool)
-			conns := tcpNode.Network().Conns()
-			for _, conn := range conns {
-				peers[conn.RemotePeer()] = true
-			}
-			log.Info(ctx, "Libp2p TCP open connections",
-				z.Int("total", len(conns)),
-				z.Int("peers", len(peers)),
-			)
-		}
-
-		log.Info(ctx, "Libp2p TCP relay started",
-			z.Str("peer_name", p2p.PeerName(tcpNode.ID())),
-			z.Any("p2p_tcp_addr", config.P2PConfig.TCPAddrs),
-		)
-
-		<-ctx.Done()
-		_ = tcpNode.Close()
-		_ = relayService.Close()
-	}()
-
-	// Start serving http
-	serverErr := make(chan error, 1)
-	go func() {
+		// Serve prometheus metrics wrapped with bootnode identifiers.
 		mux := http.NewServeMux()
-		mux.HandleFunc("/enr", func(w http.ResponseWriter, r *http.Request) {
-			_, _ = w.Write([]byte(localEnode.Node().String()))
-		})
-		server := http.Server{Addr: config.HTTPAddr, Handler: mux, ReadHeaderTimeout: time.Second}
+		mux.Handle("/metrics", promhttp.InstrumentMetricHandler(
+			promRegistry, promhttp.HandlerFor(promRegistry, promhttp.HandlerOpts{}),
+		))
+
+		// Copied from net/http/pprof/pprof.go
+		mux.HandleFunc("/debug/pprof/", pprof.Index)
+		mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+		mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+		mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+		mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+
+		server := http.Server{Addr: config.MonitoringAddr, Handler: mux, ReadHeaderTimeout: time.Second}
 		serverErr <- server.ListenAndServe()
 	}()
 
+	log.Info(ctx, "Libp2p TCP relay started",
+		z.Str("peer_name", p2p.PeerName(tcpNode.ID())),
+		z.Any("p2p_tcp_addr", config.P2PConfig.TCPAddrs),
+	)
 	log.Info(ctx, "Discv5 UDP bootnode started",
 		z.Str("p2p_udp_addr", config.P2PConfig.UDPAddr),
 		z.Str("enr", localEnode.Node().String()),
 	)
-	log.Info(ctx, "Runtime ENR available via http",
-		z.Str("url", fmt.Sprintf("http://%s/enr", config.HTTPAddr)),
-	)
+	if config.HTTPAddr != "" {
+		log.Info(ctx, "Runtime ENR available via http",
+			z.Str("url", fmt.Sprintf("http://%s/enr", config.HTTPAddr)),
```

Comment:
config.HTTPAddr should be something like this `http://abc.xyz`, this should be 
`fmt.Sprintf("%s/enr", config.HTTPAddr)` i think

Reply:
it shouldn't be:
```
	flags.StringVar(&config.HTTPAddr, "bootnode-http-address", "127.0.0.1:3640", "Listening address (ip and port) for the bootnode http server serving runtime ENR.")

```

Reply:
can add a ticket to do validation of all listen address flags in the repo if you want.